micropython/lora/lora-async/lora/async_modem.py

```diff
@@ -0,0 +1,149 @@
+        will_irq = self._modem.start_receive(None, True, rx_length)
+        return AsyncContinuousReceiver(self, will_irq, rx_packet)
+
+    async def _receive(self, will_irq, rx_packet):
+        # Second half of receiving is implemented here to share code with AsyncContinuousReceiver,
+        # until future refactor is possible (see comment immediately above this one.)
+        rx = True
+        while rx is True:
+            await self._wait(will_irq, 0, _RX_POLL_WITH_IRQ if will_irq else _RX_POLL_NO_IRQ)
+            rx = self._modem.poll_receive(rx_packet)
+        if rx:  # RxPacket instance
+            return rx
+
+    async def transmit(self, packet, tx_at_ms=None):
+        self._flags[1].clear()
+        self._modem.prepare_transmit(packet)
+
+        timeout_ms = self.get_time_on_air_us(len(packet))
+
+        if tx_at_ms is not None:
+            await uasyncio.sleep_ms(max(0, utime.ticks_diff(tx_at_ms, utime.ticks_ms())))
+
+        will_irq = self._modem.start_transmit()
+
+        tx = True
+        while tx is True:
+            await self._wait(will_irq, 1, timeout_ms)
+            tx = self._modem.poll_transmit()
+
+            # If we've already waited the estimated transmit time and the modem
+            # is not done, timeout and poll more rapidly from here on (unsure if
+            # this is necessary, outside of a serious bug, but doesn't hurt.)
+            timeout_ms = _TX_POLL_LATE_IRQ
+
+        return tx
+
+    async def _wait(self, will_irq, idx, timeout_ms):
+        # Common code path for blocking on an interrupt, if configured.
+        #
+        # idx is into _flags tuple and is 0 for rx and 1 for tx
+        #
+        # timeout_ms is the expected transmit time for transmits, or a reasonable
+        # polling timeout for receives.
+        if will_irq:
+            # In theory we don't need to ever timeout on the flag as the ISR will always
+            # fire, but this gives a workaround for possible race conditions or dropped interrupts.
+            try:
+                await uasyncio.wait_for_ms(self._flags[idx].wait(), timeout_ms)
+            except uasyncio.TimeoutError:
+                pass
+        else:
+            await uasyncio.sleep_ms(timeout_ms)
+
+    def _callback(self, _):
+        # IRQ callback from BaseModem._radio_isr. Hard IRQ context unless _DEBUG
+        # is on.
+        #
+        # Set both RX & TX flag. This isn't necessary for "real" interrupts, but may be necessary
+        # to wake both for the case of a "soft" interrupt triggered by sleep() or standby(), where
+        # both tasks need to unblock and return.
+        #
+        # The poll_receive() method is a no-op if _tx is set, so waking up both
+        # blocked tasks doesn't have much overhead (no second polling of modem
+        # IRQ flags, for example).
+        for f in self._flags:
+            f.set()
+
+    # Wrappers for common API calls on the inner modem object
```

Comment:
Unless the below methods are overridden by the SX12xx class, I think it's possible (and it's simpler, less code) to do:
```py
configure = BaseModem.configure
```

Reply:
Unfortunately these all call through to `self.modem.configure()` not `self.configure`, so I think no easier way to bind it.

(Unless this class was implemented with multiple inheritance instead, see comment below.)